internal/service/service_test.go

```diff
@@ -148,3 +148,163 @@ func TestServiceV2(t *testing.T) {
+			clusterMappingFilepath: notFoundClusterMapping,
+			expectedAnError:        true,
+			ocpVersion:             "",
+			wantConfiguration:      "",
+		},
+		{
+			name:                   "cluster mapping is undefined",
+			clusterMappingFilepath: "",
+			expectedAnError:        true,
+			ocpVersion:             "",
+			wantConfiguration:      "",
+		},
+		{
+			name:                   "valid cluster mapping and version out of range",
+			clusterMappingFilepath: validClusterMapping,
+			expectedAnError:        false,
+			ocpVersion:             "any version",
+			wantConfiguration:      configDefaultConfiguration,
+			expect400:              true,
+		},
+		{
+			name:                   "cluster version prior to 4.0",
+			clusterMappingFilepath: validClusterMapping,
+			expectedAnError:        false,
+			ocpVersion:             "1.2.3",
+			expect400:              true,
+		},
+		{
+			name:                   "cluster version is 4.0.0",
+			clusterMappingFilepath: validClusterMapping,
+			expectedAnError:        false,
+			ocpVersion:             "4.0.0",
+			wantConfiguration:      emptyConfiguration,
+		},
+		{
+			name:                   "cluster version is between 4.0.0 and 4.17.0-0",
+			clusterMappingFilepath: validClusterMapping,
+			expectedAnError:        false,
+			ocpVersion:             "4.5.6",
+			wantConfiguration:      emptyConfiguration,
+		},
+		{
+			name:                   "cluster version is a CI release of 4.17.0-0",
+			clusterMappingFilepath: validClusterMapping,
+			expectedAnError:        false,
+			ocpVersion:             "4.17.0-0.ci-2024-08-19-220527",
+			wantConfiguration:      experimental1Configuration, // TODO: or shall we return the one for versions < 4.17.0-0?
+		},
+		{
+			name:                   "cluster version is 4.17.0-0",
+			clusterMappingFilepath: validClusterMapping,
+			expectedAnError:        false,
+			ocpVersion:             "4.17.0-0",
+			wantConfiguration:      experimental1Configuration,
+		},
+		{
+			name:                   "cluster version is between 4.17.0-0 and 4.17.0",
+			clusterMappingFilepath: validClusterMapping,
+			expectedAnError:        false,
+			ocpVersion:             "4.17.0-5",
+			wantConfiguration:      experimental1Configuration,
+		},
+		{
+			name:                   "cluster version is between 4.17.0 and 4.17.5",
+			clusterMappingFilepath: validClusterMapping,
+			expectedAnError:        false,
+			ocpVersion:             "4.17.3",
+			wantConfiguration:      experimental2Configuration,
+		},
+		{
+			name:                   "cluster version is between 4.17.5 and 4.17.6",
+			clusterMappingFilepath: validClusterMapping,
+			expectedAnError:        false,
+			ocpVersion:             "4.17.6-alpha",
+			wantConfiguration:      bugWorkaroundConfiguration,
+		},
+		{
+			name:                   "cluster version is a prerelease of 4.17.6",
+			clusterMappingFilepath: validClusterMapping,
+			expectedAnError:        false,
+			ocpVersion:             "4.17.6-alpha",
```

Comment:
There's no such OCP version, but it doesn't matter :) Would it make sense to add also test for e.g `4.17.0-rc.0` or is it basically the same as `"4.17.0-0.ci-2024-08-19-220527"` :thinking: ?

Reply:
best way is to test it 😄 PTAL: https://github.com/RedHatInsights/insights-operator-gathering-conditions-service/pull/266/commits/b9a7d0040a625edda1bb143dbdbb3ba8888d3ea8 I'm not sure whether it should return `experimental1Configuration` or `emptyConfiguration` given that the cluster mapping is
```
[
    ["4.0.0", "empty.json"],
    ["4.17.0-0", "experimental_1.json"],
    ["4.17.0", "experimental_2.json"],
    ["4.17.5", "bug_workaround.json"],
    ["4.17.6", "experimental_2.json"]
]
```

WDYT @jholecek-rh ?

Reply:
with the current implementation it returns `experimental_1.json`

Reply:
never mind, I was mixing things. I made the tests more clear now: https://github.com/RedHatInsights/insights-operator-gathering-conditions-service/pull/266/commits/5a11f903411e0a506252ae5f6cf3b55f38a61a01

Reply:
`experimenta_1.json` is the expected result given the mapping above.

`4.17.0-0 <= 4.17.6-alpha < 4.17.0` in semver.

Reply:
Sorry, deleted my previous comment. I realized I wasn't sure what the question was exactly.

Reply:
https://github.com/RedHatInsights/insights-operator-gathering-conditions-service/commit/5a11f903411e0a506252ae5f6cf3b55f38a61a01 is correct for [this mapping](https://github.com/RedHatInsights/insights-operator-gathering-conditions-service/pull/266#discussion_r1732315969).

`4.17.5 <= 4.17.6-0.ci-2024-08-19-220527 < 4.17.6` in semver. Same for `4.17.6-rc.0`.